catalog/src/main/java/it/eng/catalog/rest/protocol/CatalogProtocolController.java

```diff
@@ -13,93 +13,143 @@
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.databind.ObjectMapper;
 
+import it.eng.catalog.model.Catalog;
 import it.eng.catalog.model.CatalogError;
 import it.eng.catalog.model.CatalogRequestMessage;
+import it.eng.catalog.model.Dataset;
+import it.eng.catalog.model.DatasetRequestMessage;
 import it.eng.catalog.model.Reason;
 import it.eng.catalog.model.Serializer;
 import it.eng.catalog.service.CatalogService;
+import it.eng.catalog.service.DatasetService;
 import it.eng.tools.exception.BadRequestException;
 import it.eng.tools.exception.ResourceNotFoundException;
 import it.eng.tools.model.DSpaceConstants;
 import lombok.extern.java.Log;
 
 @RestController
-@RequestMapping(consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE, 
-	path = "/catalog")
+@RequestMapping(consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE, path = "/catalog")
 @Log
 public class CatalogProtocolController {
 
 	private final CatalogService catalogService;
-	
-	public CatalogProtocolController(CatalogService catalogService) {
+	private final DatasetService datasetService;
+
+	public CatalogProtocolController(CatalogService catalogService, DatasetService datasetService) {
 		super();
 		this.catalogService = catalogService;
+		this.datasetService = datasetService;
 	}
 
+	/**
+	 * Endpoint for handling Catalog requests
+	 * 
+	 * @param authorization - Authorization header
+	 * @param body          - CatalogRequestMessage
+	 * @return - Catalog or CatalogError
+	 */
 	@PostMapping(path = "/request")
-	protected ResponseEntity<String> getCatalog(@RequestHeader(required = false) String authorization, 
-			@RequestBody JsonNode object) {
-		log.info("Handling get catalog");
+	protected ResponseEntity<String> getCatalog(@RequestHeader(required = false) String authorization,
+			@RequestBody JsonNode body) {
+		log.info("Handling catalog request");
 		verifyAuthorization(authorization);
 		try {
-			verifyRequestMessage(object);
-			var catalogRequestMessage = Serializer.deserializePlain(object.toString(), CatalogRequestMessage.class);
-//			var catalogRequestMessage = jsonToCatalogRequestMessageTransform.transform(object);
-			//TODO review how to handle filter, there should be ONE logical catalog 
-			var cat = catalogService.findAll();
-//			JsonNode jsonNode = jsonFromCatalogTransform.transform(cat);
-			
-			return ResponseEntity.ok().header("foo", "bar")
+			CatalogRequestMessage catalogRequestMessage = verifyCatalogRequestMessage(body);
```

Comment:
Created ticket to see how we are going to do sync and async calls in scope of error handling

Reply:
👍